### DISCUSSION TITLE
How to retrieve CsomServiceError from CsomServiceException?

### DISCUSSION THREAD

**Discussion by aaronmorgan** [pnp/pnpcore#976] [Q&A] [answered]
Hi, I'm trying to return a more meaningful error message when a CsomServiceException occurs and it seems like I should be able to get the Error.Message property if I treat the exception type as `CsomServiceError` but the type cannot be found.

I've done something similar for `SharePointRestServiceException` errors and used the `SharePointRestError` Error object without issue.

![image](https://user-images.githubusercontent.com/4121193/192497544-8f0932bc-19f7-4832-b47c-0c97bdab036d.png)

What's the correct way to go about this? 

![image](https://user-images.githubusercontent.com/4121193/192498074-13520d79-11dc-4b62-9dca-c102ea0ffc9e.png)

Thank you,
Aaron.

**Comment by jansenbe** [comment, marked as answer]
@aaronmorgan : the type you need is `CsomError`, not `CsomServiceError`.  The `Type` property actually is a [custom enum](https://pnp.github.io/pnpcore/api/PnP.Core.ErrorType.html), not the type of the error object.

**Reply by aaronmorgan** [reply]
Yes, of course you're right, it's now working.

Thank you,
Aaron.
